### ISSUE TITLE
1.17.1 - 37.0.27 - 5.0.0.1 - Keep Inventory Bug

### ISSUE THREAD

**Issue by Snaitf** [TheIllusiveC4/Curios#160]
#### Versions (Be specific, do not write "latest"):
* Curios: curios-forge-1.17.1-5.0.0.1.jar
* Forge: 1.17.1 - 37.0.27

#### Observed Behavior:
When the GameRule: KeepInventory is set to TRUE, all items in Curios slots are voided.  They're not even dropped on the ground upon death.

#### Expected Behavior:
Inventory should be kept when dying.

#### Steps to Reproduce:
1. Set GameRule KeepInvetory to True.
2. Place item in Curios slot.
3. Die

I've observed this behavior on both my SP and my server with MC 1.17.1, Forge 37.0.27, and Curios 5.0.0.1.
I've tested this with ONLY Curios and 1 other mod to add a slotted item.
I then tested it with ONLY Curios and 1 different mod, from a different author, to add a slotted item.
THEN I tested with ONLY Curios, and the datapack/config from the wiki walkthrough, adding an Egg slot.

In all cases the behavior was the same:
When KeepInventory is set to FALSE, all Curios slotted items are dropped upon death, as expected.
When KeepInventory is set to TRUE, all Curios slotted items are voided.

#### Crash Log: Please use a paste site like [hastebin](https://hastebin.com/).
n/a

**Comment by TheIllusiveC4** [comment]
I cannot reproduce this issue. Using your Forge and Curios versions, and setting `keepInventory` to `true`, I keep my Curios as expected. In your instance where you tested with _only_ Curios, in your custom Egg slot, what item did you place in it?

**Comment by Snaitf** [comment]
Pretty sure it was an egg.

Newest versions of Forge and Curios do not do this anymore, but I can still reproduce on the versions I listed.
Forge 27, and Curios 001.

I used only Curios and a mod called Angel Ring this time:
Created a new world
Gave myself an Angel Ring
Equipped it in Curios Ring Slot
/kill myself
-Ring dropped like normal
Picked ring back up
Reequipped it in Curios Ring Slot
Set /gamerule KeepInventory True
/kill myself
-Ring disappeared

But, like I said, it's not doing it on latest versions, so this may be moot.
